devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy

```diff
@@ -0,0 +1,683 @@
+		} else if (cmd.event == 0x03) {
+			result << createEvent(descriptionText: "$device.displayName covering was removed", isStateChange: true)
+		}
+	} else if (cmd.notificationType) {
+		def text = "Notification $cmd.notificationType: event ${([cmd.event] + cmd.eventParameter).join(", ")}"
+		result << createEvent(name: "notification$cmd.notificationType", value: "$cmd.event", descriptionText: text, displayed: false)
+	} else {
+		def value = cmd.v1AlarmLevel == 255 ? "active" : cmd.v1AlarmLevel ?: "inactive"
+		result << createEvent(name: "alarm $cmd.v1AlarmType", value: value, displayed: false)
+	}
+
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensorbinaryv2.SensorBinaryReport cmd) {
+	logTrace "SensorBinaryReport: $cmd"
+	def map = [:]
+	map.value = cmd.sensorValue ? "open" : "closed"
+	map.name = "contact"
+	if (map.value == "open") {
+		map.descriptionText = "${device.displayName} is open"
+	}
+	else {
+		map.descriptionText = "${device.displayName} is closed"
+	}
+	createEvent(map)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.indicatorv1.IndicatorReport cmd) {
+	logTrace "${cmd}"
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	logDebug "Ignored Command: $cmd"
+	return []
+}
+
+private getEventMap(name, value, displayed=null, desc=null, unit=null) {
+	def isStateChange = (device.currentValue(name) != value)
+	displayed = (displayed == null ? isStateChange : displayed)
+	def eventMap = [
+			name: name,
+			value: value,
+			displayed: displayed,
+			isStateChange: isStateChange,
+			descriptionText: desc ?: "${device.displayName} ${name} is ${value}"
+	]
+
+	if (unit) {
+		eventMap.unit = unit
+		eventMap.descriptionText = "${eventMap.descriptionText}${unit}"
+	}
+	if (displayed) {
+		logDebug "${eventMap.descriptionText}"
+	}
+	return eventMap
+}
+
+
+private wakeUpNoMoreInfoCmd() {
+	return secureCmd(zwave.wakeUpV1.wakeUpNoMoreInformation())
+}
+
+private batteryGetCmd() {
+	return secureCmd(zwave.batteryV1.batteryGet())
+}
+
+private sensorBinaryGetCmd() {
+	return secureCmd(zwave.sensorBinaryV2.sensorBinaryGet())
+}
+
+private sensorMultilevelGetCmd(sensorType) {
+	def scale = (sensorType == tempSensorType) ? 0 : 1
+	return secureCmd(zwave.sensorMultilevelV5.sensorMultilevelGet(scale: scale, sensorType: sensorType))
+}
+
+private configGetCmd(param) {
+	return secureCmd(zwave.configurationV1.configurationGet(parameterNumber: param.num))
+}
+
+private configSetCmd(param, value) {
+	return secureCmd(zwave.configurationV1.configurationSet(parameterNumber: param.num, size: param.size, scaledConfigurationValue: value))
+}
+
+private secureCmd(cmd) {
+	try {
+		if (zwaveInfo?.zw?.contains("s") || ("0x98" in device?.rawDescription?.split(" "))) {
+			return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		}
+		else {
+			return cmd.format()
+		}
+	}catch (ex) {
+		return cmd.format()
+	}
+}
+
+
+private static getCommandClassVersions() {
+	[
+			0x30: 2,  // SensorBinary
```

Comment:
Could You indent this section with 2 tabs ?

Reply:
OK, I  adjust it.